src/plugins/intel_cpu/tests/functional/utils/x64/filter_cpu_info.cpp

```diff
@@ -55,4 +55,24 @@ std::vector<CPUSpecificParams> filterCPUInfoForDevice(const std::vector<CPUSpeci
     return resCPUParams;
 }
 
+std::vector<CPUSpecificParams> filterCPUInfoForDeviceWithFP16(const std::vector<CPUSpecificParams>& allParams) {
+    std::vector<CPUSpecificParams> specificParams;
+    if (!(ov::with_cpu_x86_avx512_core_fp16())) {
+        return specificParams;
+    }
+    std::copy_if(allParams.begin(), allParams.end(), std::back_inserter(specificParams), [](const CPUSpecificParams& item) {
+        const auto &selected = std::get<3>(item);
+        if ((!ov::with_cpu_x86_avx512_core_amx_fp16()) && selected.find("amx") != std::string::npos) {
+            return false;
+        }
+        if (selected.find("avx2") != std::string::npos
+                || (selected.find("avx") != std::string::npos && selected.find("avx512") == std::string::npos)
+                || selected.find("sse42") != std::string::npos) {
+            return false;
```

Comment:
these logic is  wordy, why not filter out everything except "amx"+`ov::with_cpu_x86_avx512_core_amx_fp16()` and "avx512"+ `ov::with_cpu_x86_avx512_core_fp16()`?

Reply:
ok，I will update it

Reply:
have already updated it

Reply:
@usstq help to continue review thanks